Improved parameter handling by continuous distributions

supreet11agrawal opened the pull request:
This improves the parametric handling by different distributions.
Many distributions have specific requirements for parameters which were
not implemented in Sympy.
Also added test cases for checking if further changes don't break the
specific requirements.

<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests . Please also
write a comment on that issue linking back to this pull request once it is
open. -->
Finishes #16159 

#### Brief description of what is fixed or changed


#### Other comments


#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* stats
  * Improved parameter handling by continuous distributions
<!-- END RELEASE NOTES -->

supreet11agrawal committed:
Improved parameter handling by continuous distributions

This improves the parametric handling by different distributions.
Many distributions have specific requirements for parameters which were
not implemented in Sympy.
Also added test cases for checking if further changes don't break the
specific requirements.

jksuom commented on sympy/stats/crv_types.py:
```diff
@@ -922,7 +922,7 @@ def Exponential(name, rate):
     ==========
 
-    rate : A positive Real number, `\lambda > 0`, the rate (or inverse scale/inverse mean)
+    lambda : A positive Real number, `\lambda > 0`, the rate (or inverse scale/inverse mean)
```
`lambda` is a keyword. It cannot be used as an identifier.

supreet11agrawal replied:
Actually in https://docs.sympy.org/latest/modules/stats.html?highlight=stats#module-sympy.stats.crv_types 
for every distribution, parameters are defined in the way  `Symbol : Property/ies Name`

Eg.
https://github.com/sympy/sympy/blob/d2fcf9ecdfef464b26f0e118018a7817059d22a2/sympy/stats/crv_types.py#L271-L273
https://github.com/sympy/sympy/blob/d2fcf9ecdfef464b26f0e118018a7817059d22a2/sympy/stats/crv_types.py#L357-L358
https://github.com/sympy/sympy/blob/d2fcf9ecdfef464b26f0e118018a7817059d22a2/sympy/stats/crv_types.py#L502-L503
The symbols are in accordance with the distribution function written. I have made this change in that manner only.

supreet11agrawal commented:
Kindly add Stats label

supreet11agrawal committed:
Changed the parameter name to l

jksuom commented on sympy/stats/crv_types.py:
```diff
@@ -922,7 +922,7 @@ def Exponential(name, rate):
     ==========
 
-    rate : A positive Real number, `\lambda > 0`, the rate (or inverse scale/inverse mean)
+    l : A positive Real number, `\lambda > 0`, the rate (or inverse scale/inverse mean)
```
This should be the same as in the function definition of `Exponential` and in `_argnames` of `ExponentialDistribution`.

supreet11agrawal replied:
I have restored it back to `rate`

supreet11agrawal committed:
Restored the parameter name back to rate

Upabjojr commented on sympy/stats/crv_types.py:
```diff
@@ -956,16 +956,8 @@ def Exponential(name, rate):
     >>> skewness(X)
     2
 
-    >>> X = Exponential('x', 10)
```
why has this been removed?

supreet11agrawal replied:
The idea was to show the precise working of the exponential distribution. Substituting a number in place of `lambda` was obvious and increased the length of docs unnecessarily. Examples are shown with the help of a general parameter `l`.
Complete properties of Exponential distribution are described in form of a general parameter `lambda`.
https://github.com/sympy/sympy/blob/f552493265c8f15695f8254eb1812873c25f33e3/sympy/stats/crv_types.py#L942-L960

Upabjojr replied:
I don't see the point of removing documentation code. A few more examples are OK.

Upabjojr commented:
Did you change the definitions of the distributions by chance? It is usually a good idea to create short PRs, they are easier to review.

supreet11agrawal commented:
> Did you change the definitions of the distributions by chance? It is usually a good idea to create short PRs, they are easier to review.

No, I haven't changed the definitions of distributions. However, I removed some minor errors from some distributions. For eg in PR #16159, I changed the Cauchy distribution for which cdf was written in docs instead of pdf.
I'll keep in mind to make small PRs next time.
Thanks

jksuom commented on sympy/stats/crv_types.py:
```diff
@@ -1353,8 +1360,8 @@ def GammaInverse(name, a, b):
 
-    a : Real number, `a > 0` a shape
-    b : Real number, `b > 0` a scale
+    alpha : Real number, `\alpha > 0` a shape
```
This should agree with the call parameter of GammaInverse and _argnames of GammaInverseDistribution.

supreet11agrawal replied:
Ok, I'll change it

supreet11agrawal committed:
Improved consistency in parameters of some distributions

jksuom commented on sympy/stats/crv_types.py:
```diff
@@ -936,10 +936,10 @@ def Exponential(name, rate):
     >>> from sympy import Symbol
 
-    >>> l = Symbol("lambda", positive=True)
+    >>> rate = Symbol("lambda", positive=True)
```
It was not necessary to change this name in Examples though it does no harm.

supreet11agrawal replied:
Just for consistency

Upabjojr commented on sympy/stats/crv_types.py:
```diff
@@ -910,7 +910,7 @@ def _moment_generating_function(self, t):
 def Exponential(name, rate):
     r"""
-    Create a continuous random variable with an Exponential distribution.
+    Creates a Continuous Random Variable with Exponential distribution.
```
I don't understand why you want to change this sentence. If you use `creates` you should add the subject first, e.g. _This function creates ..._. It makes sense to start the sentence with _Create_ but not with _Creates_.

Upabjojr replied:
I would leave these sentences as they are.

Upabjojr commented:
I would suggest to close this PR and re-open a new PR with only the necessary edits. There are lots of useless changes that make it difficult to review this PR.

supreet11agrawal commented:
Ok, I'll close this PR and submit PRs distribution wise. Now, what changes should I keep? @Upabjojr @jksuom
